### PR TITLE
(jkube-kit/resource/service/) : Migrate tests from JMockit to Mockito

### PR DESCRIPTION
**Pull request by baruKreddy**
Fixes https://github.com/eclipse/jkube/issues/1604

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [ ] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] My code follows the style guidelines of this project
 - [ ] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by baruKreddy** [comment]
@rohanKanojia Making the PR as draft, as I have commented the verification call as testcase is failing when we pass null argument.

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java [hunk 1]
```diff
@@ -120,7 +133,7 @@ void writeResourcesIndividualAndComposite_withResourcesWithSameName_shouldWriteA
     // When
     WriteUtil.writeResourcesIndividualAndComposite(klb.build(), resourceFileBase, null, log);
     // Then
-    verifyResourceUtilSave(null, 6);
+    //verifyResourceUtilSave(null, 6);
```
Comment (rohanKanojia) [inline jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java:136]: I checked on master branch. Even if I comment this test line on master branch all tests pass. While setting a breakpoint in WriteUtil.java:91 I can see that `resourceFileBase` is never passed as `null`.

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java [hunk 2]
```diff
@@ -101,7 +114,7 @@ void writeResourcesIndividualAndCompositeWithResourcesShouldWriteAll() throws IO
     // When
     WriteUtil.writeResourcesIndividualAndComposite(klb.build(), resourceFileBase, null, log);
     // Then
-    verifyResourceUtilSave(null, 3);
+    //resourceUtil.verify(() -> ResourceUtil.save(isNull(), any(), isNull()),times(3));
```
Comment (rohanKanojia) [inline jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java:117]:
```suggestion
```
https://github.com/eclipse/jkube/pull/1759/files#r976420759

File: jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java (hunk 1 above)
Comment (baruKreddy) [reply jkube-kit/resource/service/src/test/java/org/eclipse/jkube/kit/resource/service/WriteUtilTest.java:136]: Great, Thanks Rohan on confirming

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]

**Pull request merged**